### ISSUE TITLE
[feature] set MacOS deployment target in AutoToolsBuildEnvironment helper

### ISSUE THREAD

**Issue by SSE4** [conan-io/conan#7578]
following discussion in https://github.com/conan-io/conan-center-index/pull/2508

right now, recipe authors have to add the following code:
```
        if tools.is_apple_os(self.settings.os) and self.settings.get_safe("os.version"):
            target = tools.apple_deployment_target_flag(self.settings.os, self.settings.os.version)
            env_build.flags.append(target)
```
this seems to be vital and required for almost every recipe to correctly respect the `os.version` setting. it would be nice if it could be done automatically.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by SSE4** [comment]
@memsharded @jgsogo can you close this one? already done in #7862

**Comment by memsharded** [comment]
Closed by #7862
